Stellar-contract.x

```diff
@@ -255,6 +258,63 @@ struct Int128Parts {
     uint64 hi;
 };
 
+struct ContractInvocation
+{
+    Hash contractID;
+    SCSymbol functionName;
+};
+
+struct AuthorizedInvocation
+{
+    ContractInvocation callStack<>;
+    SCVec topArgs;
+    uint64* nonce;
+};
+
+enum SCAccountIdType
+{
+    SC_ACCOUNT_ID_TYPE_BUILTIN_INVOKER = 0,
+    SC_ACCOUNT_ID_TYPE_BUILTIN_CLASSIC_ACCOUNT = 1,
+    SC_ACCOUNT_ID_TYPE_BUILTIN_ED25519 = 2,
+    SC_ACCOUNT_ID_TYPE_GENERIC_ACCOUNT = 3
+};
+
+union ScAccountId switch (SCAccountIdType type)
+{
+case SC_ACCOUNT_ID_TYPE_BUILTIN_CLASSIC_ACCOUNT:
+    AccountID accountId;
+case SC_ACCOUNT_ID_TYPE_BUILTIN_ED25519:
+    Hash ed25519PublicKey;
+case SC_ACCOUNT_ID_TYPE_BUILTIN_INVOKER:
+    void;
+case SC_ACCOUNT_ID_TYPE_GENERIC_ACCOUNT:
+    Hash accountContractId;
+};
+
+struct SCAccount
+{
+    ScAccountId accountId;
+    AuthorizedInvocation invocations<>;
+    SCVec signatureArgs;
+};
+
+enum SCAddressType
+{
+    SC_ADDRESS_TYPE_CLASSIC_ACCOUNT = 0,
+    SC_ADDRESS_TYPE_ED25519 = 1,
+    SC_ADDRESS_TYPE_CONTRACT = 2
+};
+
+union SCAddress switch (SCAddressType type)
```

Comment:
Yeah, but I'm getting rid of this in favor of having `ScAddress` everywhere. The sole purpose of `ScAccountId` was the invoker support, but it can be supported differently.